shared-bindings/usb_hid/Device.c

```diff
@@ -81,82 +89,144 @@ STATIC mp_obj_t usb_hid_device_make_new(const mp_obj_type_t *type, size_t n_args
-    uint8_t report_id_index = 0;
-    if (report_id_index_arg != mp_const_none) {
-        const mp_int_t report_id_index_int = mp_obj_int_get_checked(report_id_index_arg);
-        if (report_id_index_int <= 0 || (uint32_t)report_id_index_int >= descriptor_bufinfo.len) {
-            mp_raise_ValueError_varg(translate("%q must be None or between 1 and len(report_descriptor)-1"),
-                MP_QSTR_report_id_index);
-        }
-        report_id_index = report_id_index_int;
+
+    uint8_t report_ids_array[report_ids_count];
+    uint8_t in_report_lengths_array[report_ids_count];
+    uint8_t out_report_lengths_array[report_ids_count];
+
+    // Validate the ids and lengths are all integers in range.
+    for (size_t i = 0; i < report_ids_count; i++) {
+        mp_obj_t i_obj = MP_OBJ_NEW_SMALL_INT(i);
+
+        report_ids_array[i] = (uint8_t)mp_arg_validate_int_range(
+            // It's not the actual argument that's out of range, but its elements.
+            // But the error message is close enough.
+            MP_OBJ_SMALL_INT_VALUE(mp_obj_subscr(report_ids, i_obj, MP_OBJ_SENTINEL)),
+            1, 255, MP_QSTR_report_ids);
+
+        in_report_lengths_array[i] = (uint8_t)mp_arg_validate_int_range(
+            MP_OBJ_SMALL_INT_VALUE(mp_obj_subscr(in_report_lengths_array, i_obj, MP_OBJ_SENTINEL)),
+            0, 255, MP_QSTR_in_report_lengths);
+
+        out_report_lengths_array[i] = (uint8_t)mp_arg_validate_int_range(
+            MP_OBJ_SMALL_INT_VALUE(mp_obj_subscr(out_report_lengths_array, i_obj, MP_OBJ_SENTINEL)),
+            0, 255, MP_QSTR_out_report_lengths);
     }
 
     common_hal_usb_hid_device_construct(
-        self, descriptor, usage_page, usage, in_report_length, out_report_length, report_id_index);
+        self, descriptor, usage_page, usage, report_ids_count, report_ids_array, in_report_lengths_array, out_report_lengths_array);
     return (mp_obj_t)self;
 }
 
 
-//|     def send_report(self, buf: ReadableBuffer) -> None:
-//|         """Send a HID report."""
+//|     def send_report(self, buf: ReadableBuffer, report_id: Optional[int] = None) -> None:
+//|         """Send an HID report. If the device descriptor specifies zero or one report id's,
+//|         you can supply `None` (the default) as the value of ``report_id``.
+//|         Otherwise you must specify which report id to use when sending the report.
+//|         """
 //|         ...
 //|
-STATIC mp_obj_t usb_hid_device_send_report(mp_obj_t self_in, mp_obj_t buffer) {
-    usb_hid_device_obj_t *self = MP_OBJ_TO_PTR(self_in);
+STATIC mp_obj_t usb_hid_device_send_report(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
+    usb_hid_device_obj_t *self = MP_OBJ_TO_PTR(pos_args[0]);
+
+    enum { ARG_buf, ARG_report_id };
+    static const mp_arg_t allowed_args[] = {
+        { MP_QSTR_buf, MP_ARG_REQUIRED | MP_ARG_OBJ },
+        { MP_QSTR_report_id, MP_ARG_OBJ, {.u_obj = mp_const_none} },
```

Comment:
Why not just default this to -1 instead of None?

Reply:
I did think about doing that, but I want the user-facing API to be `None`, for clarity: `None` means no report id supplied. If the user does supply the parameter (instead of defaulting), it should still work the same way.